### PR TITLE
RavenDB-7203 Database record isn't updated after node removal

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs [hunk 1]
```diff
@@ -73,19 +73,21 @@ public Task GetClusterTopology()
 
                 if (topology.Members.Count == 0)
                 {
-                    var serverUrl = GetStringQueryString("url");
+                    //var serverUrl = GetStringQueryString("url");
+                    var serverUrl = ServerStore.RavenServer.WebUrls[0];
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs]: Why?

File: src/Raven.Server/ServerWide/ClusterStateMachine.cs [hunk 2]
```diff
@@ -567,6 +568,11 @@ public DatabaseRecord ReadDatabase(TransactionOperationContext context, string n
             var doc = Read(context, "db/" + name.ToLowerInvariant(), out etag);
             if (doc == null)
                 return null;
+            // If we are in passive state, we have prevent from tasks to be performed by this node.
+            if (_parent.CurrentState == RachisConsensus.State.Passive)
```
Comment (ayende) [inline src/Raven.Server/ServerWide/ClusterStateMachine.cs]: That is extremely invasive operation here. You are basically saying that while the node is passive, no databases can be accessed.
If I just remove a node from the cluster, I lost access to all its data.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/ServerWide/ClusterStateMachine.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Server/ServerWide/ClusterStateMachine.cs]: This block the access to the database state, not the database itself.
I'll still be able to access the storage of the data, indexes, etc.

Comment (ayende) [reply src/Raven.Server/ServerWide/ClusterStateMachine.cs]: Not a good idea, I think.
This is lying to the rest of the system.
And it won't help you with the state change for already running stuff, I think.

You need to be active about this and check in all the relevant locations.

**Pull request closed without merging**